website/docs/advanced-concepts/embed-appsmith-into-existing-application.mdx

```diff
@@ -58,135 +50,61 @@ In self-hosted instances, you can restrict embedding an application to certain w
 
-<Tags
-  tags={[
-    {
-      name: "Business",
-      link: "https://www.appsmith.com/pricing",
-      additionalClass: "business",
-    }
-  ]}
-/>
-</div>
-
-<!-- vale on -->
-
-![](/img/postmessage_parent_incoming.png)
-
-When you embed an Appsmith app as an iframe on a website, the event listeners allow you to listen to the message from that parent website. You can use this method to make Appsmith react to events from the parent website.
-
-On your Appsmith app, you can enable/disable a page to react to these messages using the following functions -
-
-- [windowMessageListener()](/reference/appsmith-framework/widget-actions/window-message-listener)
-- [unlistenWindowMessage()](/reference/appsmith-framework/widget-actions/unlisten-window-message)
-
-For example, a parent website (`https://mywebsite.com`) where an appsmith app is embedded calls this function when a button is clicked -
-
-```javascript
-const iFrame = document.getElementById(”#appsmith-iframe”);
-iFrame.contentWindow.postMessage("Parent message", 'https://your-appsmith-domain.com');
-```
-
-In the Appsmith app, if you want to run an API called `Api1` in reaction to this message, you can use the `windowMessageListener()` function as follows -
-
-```javascript
-windowMessageListener(”https://mywebsite.com”, () => Api1.run());
-```
-
-To stop the Appsmith app from reacting to the incoming messages from the parent website (`https://mywebsite.com`), you can use the `unlistenWindowMessage` method as follows -
-
-```javascript
-unlistenWindowMessage(”https://mywebsite.com”)
-```
+### Prerequisites
 
-:::tip
-You can automatically set up an action on a page by calling the `windowMessageListener` in a JS object method and have it run when the page loads.
-:::
+* Ensure you have a self-hosted instance of Appsmith. See the [Appsmith installation guides](/getting-started/setup/installation-guides) for detailed instructions on setting up your Appsmith instance.
+* Ensure that Frame ancestors include the parent app domain if you have restricted access for embedding Appsmith. For more information, see [Frame ancestors](/getting-started/setup/instance-configuration/frame-ancestors).
+* Ensure that you've hosted Appsmith app under the sub-domain of the same domain as the parent app. For example, `appsmith.company.com` and `internal.company.com` sharing the same domain (`company.com`).
+
+### Embed Appsmith
+
+1. Set up your Single Sign-On (SSO) provider on your Appsmith instance. See the [SSO configuration guides](/getting-started/setup/instance-configuration/authentication) for detailed steps on configuring your SSO provider. When configuring **Google OAuth 2.0**, configure the parent and child as two OAuth 2.0 clients under the same project.
+2. Go to the **App Settings** > **Share & Embed** within your Appsmith application.
+3. Copy the provided **Embed URL**.
```

Comment:
We should have this line before copying URL

select the preferred SSO type from the SSO method list.

Reply:
For me when I configured the SSO on Appsmith, the method was auto-selected. And hence, I've not added it.